File: app/src/main/java/com/example/lpiem/rickmorty_tcg/model/Info.java
package com.example.lpiem.rickmorty_tcg.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Info implements Parcelable
{

    @SerializedName("count")
    @Expose
    private Integer count;
    @SerializedName("pages")
    @Expose
    private Integer pages;
    @SerializedName("next")
    @Expose
    private String next;
    @SerializedName("prev")
    @Expose
    private String prev;
    public final static Parcelable.Creator<Info> CREATOR = new Creator<Info>() {


        @SuppressWarnings({
                "unchecked"
        })
        public Info createFromParcel(Parcel in) {
            return new Info(in);
        }

        public Info[] newArray(int size) {
            return (new Info[size]);
        }

    }
            ;

    protected Info(Parcel in) {
        this.count = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.pages = ((Integer) in.readValue((Integer.class.getClassLoader())));
        this.next = ((String) in.readValue((String.class.getClassLoader())));
        this.prev = ((String) in.readValue((String.class.getClassLoader())));
    }

    public Info() {
    }

    public Integer getCount() {
        return count;
    }

    public void setCount(Integer count) {
        this.count = count;
    }

    public Integer getPages() {
        return pages;
    }

    public void setPages(Integer pages) {
        this.pages = pages;
    }

    public String getNext() {
        return next;
    }

    public void setNext(String next) {
        this.next = next;
    }

    public String getPrev() {
        return prev;
    }

    public void setPrev(String prev) {
        this.prev = prev;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(count);
        dest.writeValue(pages);
        dest.writeValue(next);
        dest.writeValue(prev);
    }

    public int describeContents() {
        return  0;
    }

}

File: RickAndMortyAPP/app/src/main/java/com/example/lpiem/rickandmortyapp/Model/Info.kt
package com.example.lpiem.rickandmortyapp.Model

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Info : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Info> = object : Parcelable.Creator<Info> {


            override fun createFromParcel(inside: Parcel): Info {
                return Info(inside)
            }

            override fun newArray(size: Int): Array<Info?> {
                return arrayOfNulls(size)
            }

        }
    }

    @SerializedName("count")
    @Expose
    var count: Int? = null
    @SerializedName("pages")
    @Expose
    var pages: Int? = null
    @SerializedName("next")
    @Expose
    var next: String? = null
    @SerializedName("prev")
    @Expose
    var prev: String? = null

    constructor(inside: Parcel) {
        this.count = inside.readValue(Int::class.java.classLoader) as Int
        this.pages = inside.readValue(Int::class.java.classLoader) as Int
        this.next = inside.readValue(String::class.java.classLoader) as String
        this.prev = inside.readValue(String::class.java.classLoader) as String
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(count)
        dest.writeValue(pages)
        dest.writeValue(next)
        dest.writeValue(prev)
    }

    override fun describeContents(): Int {
        return 0
    }



}

